packages/mirinae/src/hooks/context-menu-fixed-style/index.ts

```diff
@@ -120,6 +102,27 @@ export const useContextMenuFixedStyle = ({ useFixedMenuStyle, visibleMenu, targe
         observer.observe(contextMenuFixedStyleState.targetElement);
     });
 
+    const prevX = ref(0);
+    const prevY = ref(0);
+    const observeElementChanges = () => {
+        if (!state.useFixedMenuStyle) return;
+
+        const { x, y } = contextMenuFixedStyleState.targetElement.getBoundingClientRect();
+
+        if (x !== prevX.value || y !== prevY.value) {
+            hideMenu();
+        }
+
+        prevX.value = x;
+        prevY.value = y;
+
+        requestAnimationFrame(observeElementChanges);
```

Comment:
Reason for using `requestAnimationFrame`
- Because the goal was to obtain real-time `x` and `y` coordinates of a simple element, it was chosen to execute tasks in sync with the browser's rendering cycle using `requestAnimationFrame`, making it relatively straightforward compared to calling a callback every time DOM changes are detected and processed.